Some tests broken on 2022 R1 since recent commit

### Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### Description of the bug

Some tests are failing with 2022 R1 starting at 879c56314d264d3ada0758e6b4fbe486cdc1a77e

```
./_unittest/test_41_3dlayout_modeler.py::TestClass::test_01_creatematerial Failed with Error: [undefined]failed on setup with "IndexError: list index out of range"
self = <class '_unittest.test_41_3dlayout_modeler.TestClass'>

    def setup_class(self):
        BasisTest.my_setup(self)
>       self.aedtapp = BasisTest.add_app(self, project_name=test_project_name, application=Hfss3dLayout)

_unittest\test_41_3dlayout_modeler.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_unittest\conftest.py:129: in add_app
    self.desktop = Desktop(desktop_version, settings.non_graphical, new_thread)
pyaedt\generic\design_types.py:479: in Desktop
    return app(
pyaedt\desktop.py:418: in __init__
    self._init_cpython(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyaedt.desktop.Desktop object at 0x000002BF4F585E20>
non_graphical = True, new_aedt_session = True
version = 'Ansoft.ElectronicsDesktop.2022.1', student_version = False
version_key = '2022.1', aedt_process_id = None

    def _init_cpython(
        self,
        non_graphical,
        new_aedt_session,
        version,
        student_version,
        version_key,
        aedt_process_id=None,
    ):
        import pythoncom
    
        from pyaedt.generic.clr_module import _clr
    
        if os.name == "posix":
            raise Exception(
                "PyAEDT supports COM initialization in Windows only. To use in Linux, upgrade to AEDT 2022 R2 or later."
            )
        base_path = self._main.sDesktopinstallDirectory
        sys.path.append(base_path)
        sys.path.append(os.path.join(base_path, "PythonFiles", "DesktopPlugin"))
        launch_msg = "AEDT installation Path {}.".format(base_path)
        self.logger.info(launch_msg)
        _clr.AddReference("Ansys.Ansoft.CoreCOMScripting")
        AnsoftCOMUtil = __import__("Ansys.Ansoft.CoreCOMScripting")
        self.COMUtil = AnsoftCOMUtil.Ansoft.CoreCOMScripting.Util.COMUtil
        self._main.COMUtil = self.COMUtil
        StandalonePyScriptWrapper = AnsoftCOMUtil.Ansoft.CoreCOMScripting.COM.StandalonePyScriptWrapper
        self.logger.info("Launching AEDT with module PythonNET.")
        processID = []
        if IsWindows:
            processID = com_active_sessions(version, student_version, non_graphical)
        if student_version and not processID:  # Opens an instance if processID is an empty list
            self._run_student()
        elif non_graphical or new_aedt_session or not processID:
            # Force new object if no non-graphical instance is running or if there is not an already existing process.
            StandalonePyScriptWrapper.CreateObjectNew(non_graphical)
        else:
            StandalonePyScriptWrapper.CreateObject(version)
        processID2 = []
        if IsWindows:
            processID2 = com_active_sessions(version, student_version, non_graphical)
        proc = [i for i in processID2 if i not in processID]  # Looking for the "new" process
        if (
            not proc and (not new_aedt_session) and aedt_process_id
        ):  # if it isn't a new aedt session and a process ID is given
            proc = [aedt_process_id]
        elif not proc:
            proc = processID2
        if proc == processID2 and len(processID2) > 1:
            self._dispatch_win32(version)
        elif version_key >= "2021.2":
            if student_version:
                self.logger.info("{} Student version started with process ID {}.".format(version, proc[0]))
            else:
>               self.logger.info("{} Started with process ID {}.".format(version, proc[0]))
E               IndexError: list index out of range

pyaedt\desktop.py:684: IndexError

```

### Steps To Reproduce

- Update and pull latest main branch
- Edit _unittest/launch_config.json to run tests in 2022 R1
- Run the _unittest/test_41_3dlayout_modeler.py::TestClass::test_01_creatematerial test

### Which Operating System are you using?

Windows

### Which Python version are you using?

3.9

### Installed packages

aiohttp==3.8.3
aiosignal==1.3.1
appdirs==1.4.4
async-timeout==4.0.2
attrs==22.1.0
certifi==2022.9.24
cffi==1.15.1
charset-normalizer==2.1.1
clr-loader==0.2.4
colorama==0.4.6
contourpy==1.0.6
cycler==0.11.0
exceptiongroup==1.0.1
fonttools==4.38.0
frozenlist==1.3.3
idna==3.4
imageio==2.22.4
iniconfig==1.1.1
kiwisolver==1.4.4
matplotlib==3.6.2
multidict==6.0.2
numpy==1.23.4
packaging==21.3
pandas==1.5.1
Pillow==9.3.0
pluggy==1.0.0
plumbum==1.8.0
pooch==1.6.0
psutil==5.9.4
-e git+https://github.com/pyansys/pyaedt.git@7cf0261186282969d9eed7700f4b2233a8f91f62#egg=pyaedt
pycparser==2.21
pyparsing==3.0.9
pytest==7.2.0
python-dateutil==2.8.2
pythonnet==3.0.1
pytz==2022.6
pyvista==0.37.0
pywin32==305
requests==2.28.1
rpyc==5.3.0
scooby==0.7.0
six==1.16.0
tomli==2.0.1
urllib3==1.26.12
vtk==9.2.2
wslink==1.9.1
yarl==1.8.1

--- a/pyaedt/generic/general_methods.py
+++ b/pyaedt/generic/general_methods.py
@@ -922,14 +922,22 @@ def com_active_sessions(version=None, student_version=False, non_graphical=False
         keys = ["ansysedtsv.exe"]
     else:
         keys = ["ansysedt.exe"]
+    long_version = None
+    if len(version) > 6:
+        version = version[-6:]
     if version and "." in version:
+        long_version = version
         version = version[-4:].replace(".", "")
-    if version < "222":
+    if version < "221":
         version = version[:2] + "." + version[2]
+        long_version = "20{}".format(version)
     sessions = []
     for p in psutil.process_iter():
         try:
             if p.name() in keys:
+                if long_version and _check_installed_version(os.path.dirname(p.exe()), long_version):
+                    sessions.append(p.pid)
+                    continue
                 cmd = p.cmdline()
                 if non_graphical and "-ng" in cmd or not non_graphical:
                     if not version or (version and version in cmd[0]):
@@ -1391,6 +1399,34 @@ def _dim_arg(value, units):
         return str(val) + units
     except:
         return value
+
+
+@pyaedt_function_handler()
+def _check_installed_version(install_path, long_version):
+    """Check installation folder to determine if it is for specified Ansys EM version.
+
+    Parameters
+    ----------
+    install_path: str
+        Installation folder to check.  For example, ``"C:\\Program Files\\AnsysEM\\v231\\Win64"``.
+    long_version: str
+        Long form of version number.  For example, ``"2023.1"``.
+
+    Returns
+    -------
+    bool
+
+    """
+    product_list_path = os.path.join(install_path, "config", "ProductList.txt")
+    if os.path.isfile(product_list_path):
+        try:
+            with open(product_list_path, "r") as f:
+                install_version = f.readline().strip()[-6:]
+                if install_version == long_version:
+                    return True
+        except:
+            pass
+    return False
 
 
 class Settings(object):

